ams/server/src/main/java/com/netease/arctic/server/optimizing/TaskRuntime.java

```diff
@@ -299,10 +299,17 @@ public String toString() {
 
   private void validThread(OptimizerThread thread) {
     if (token == null) {
-      throw new IllegalStateException("Task not scheduled yet, taskId:" + taskId);
+      throw new TaskRuntimeException("Task has been reset or not yet scheduled, taskId:%s", taskId);
     }
-    if (!thread.getToken().equals(getToken()) || thread.getThreadId() != threadId) {
-      throw new DuplicateRuntimeException("Task already acked by optimizer thread + " + thread);
+    if (!thread.getToken().equals(getToken())) {
```

Comment:
I believe displaying both the token and thread ID in the error logs would facilitate problem localization more conveniently. Therefore, I suggest making some changes like:
```
    if (!thread.getToken().equals(getToken()) || thread.getThreadId() != threadId) {
      throw new TaskRuntimeException(
          "The optimizer threads do not match, the thread in the task is OptimizerThread(token=%s, threadId=%s), and the thread in the request is OptimizerThread(token=%s, threadId=%s).",
          getToken(), threadId, thread.getToken(), thread.getThreadId());
    }
```